### DISCUSSION TITLE
要怎么展开变量呢？

### DISCUSSION THREAD

**Discussion by shuax** [xmake-io/xmake#3894] [Q&A] [answered]
![image](https://github.com/xmake-io/xmake/assets/6940583/1151fe98-cb7b-4bfd-a24e-11a3b8a67864)
因为vcpkg的包有冲突，所以我不打算使用全局的vcpkg，我在本地目录引入了一个vcpkg的submodule。

然后我在xmake.lua第一行写了
`set_config("vcpkg", "$(curdir)/vcpkg")`好像不能用
但是我写死路径
`set_config("vcpkg", "E:/code/xxx/vcpkg")`就能用了。

有什么办法可以手动把`$(curdir)`提出来吗，类似`vformat`。

**Comment by waruqi** [comment, marked as answer]
set_config 里面不支持。 而且不建议用 curdir。。走 `path.join(os.scriptdir(), "xxx")`

**Reply by shuax** [reply]
牛逼。可以了。
